lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -309,11 +315,34 @@ bool SslClientConnection::connectSocket() {
     return false;
   }
 
+  if (_isSocketNonBlocking) {
+#if defined(__linux__) || defined(__APPLE__)
```

Comment:
It feels really weird that a part of the work is now done in `setSocketToNonBlocking()` and a part is done here.
It would be a lot more sensible to move this part into `setSocketToNonBlocking()` as well.